src/I18n/TranslatorRegistry.php

```diff
@@ -62,25 +81,46 @@ public function get($name, $locale = null)
             $locale = $this->getLocale();
         }
 
-        if (!isset($this->registry[$name][$locale])) {
-            $key = "translations.$name.$locale";
-            $translator = Cache::remember($key, function () use ($name, $locale) {
-                try {
-                    return parent::get($name, $locale);
-                } catch (\Aura\Intl\Exception $e) {
-                }
+        if (isset($this->registry[$name][$locale])) {
+            return $this->registry[$name][$locale];
+        }
 
-                if (!isset($this->_loaders[$name])) {
-                    $this->registerLoader($name, $this->_partialLoader());
-                }
+        if (!$this->_cacher) {
+            return $this->registry[$name][$locale] = $this->_getTranslator($name, $locale);
+        }
+
+        $key = "translations.$name.$locale";
+        $translator = $this->_cacher->read($key);
+        if (!$translator) {
+            $translator = $this->_getTranslator($name, $locale);
+            $this->_cacher->write($key, $translator);
+        }
 
-                return $this->_getFromLoader($name, $locale);
-            }, '_cake_core_');
+        return $this->registry[$name][$locale] = $translator;
+    }
+
+    /**
+     * Gets a translator from the registry by package for a locale.
+     *
+     * @param string $name The translator package to retrieve.
+     * @param string|null $locale The locale to use; if empty, uses the default
+     * locale.
+     * @return \Aura\Intl\TranslatorInterface A translator object.
+     * @throws \Aura\Intl\Exception If no translator with that name could be found
+     * for the given locale.
+     */
+    protected function _getTranslator($name, $locale)
+    {
+        try {
+            return parent::get($name, $locale);
+        } catch (\Aura\Intl\Exception $e) {
+        }
```

Comment:
There should be no try/catch here; method signature says it throws exceptions.

Also, since this is new code; should the method name have an `_` prefix?